audio/decoders/voc.cpp

```diff
@@ -431,25 +431,25 @@ void VocStream::preProcess() {
 	_length = Timestamp(0, _rate);
 
 	// Calculate the total play time and do some more sanity checks
-	for (BlockList::const_iterator i = _blocks.begin(), end = _blocks.end(); i != end; ++i) {
+	for (auto &curBlock : _blocks) {
```

Comment:
Should instances of const_iterator become const auto & in the loops?
I'm unsure of best practices for auto, but this stood out to me.

Reply:
You are right that `const auto &` enforces const. [Here](https://stackoverflow.com/questions/5070521/does-const-auto-have-any-meaning)'s a relevant SO link.

I've force-pushed the relevant changes for `const_iterator` -> `const auto &`